roxie/ccd/ccdfile.cpp

```diff
@@ -1270,6 +1616,211 @@ class CRoxieFileCache : implements IRoxieFileCache, implements ICopyFileProgress
         }
     }
 
+    static unsigned __int64 readPage(const char * &_t)
+    {
+        const char *t = _t;
+        unsigned __int64 v = 0;
+        for (;;)
+        {
+            char c = *t;
+            if ((c >= '0') && (c <= '9'))
+                v = v * 16 + (c-'0');
+            else if ((c >= 'a') && (c <= 'f'))
+                v = v * 16 + (c-'a'+10);
+            else if ((c >= 'A') && (c <= 'F'))
+                v = v * 16 + (c-'A'+10);
+            else
+                break;
+            t++;
+        }
+        _t = t;
+        return v;
+    }
+
+    virtual void loadSavedOsCacheInfo() override
+    {
+        Owned<const ITopologyServer> topology = getTopology();
+        for (unsigned channel : topology->queryChannels())
+            doLoadSavedOsCacheInfo(channel);
+        doLoadSavedOsCacheInfo(0);  // MORE - maybe only if I am also a server?
+    }
+
+    void doLoadSavedOsCacheInfo(unsigned channel)
+    {
+        const char* dllserver_root = getenv("HPCC_DLLSERVER_PATH");
+        assertex(dllserver_root != nullptr);
```

Comment:
for clarity, wouldn't it be better to introduce a HPCC_PAGECACHE_PATH or similar, so as not refer to 'dll' here?

Reply:
Maybe though that would then mean we needed ways to set it to somewhere other than the dllserver location. I think maybe better would be to add a comment saying WHY this is being stored in the dllserver location (and consider renaming that location to something else if we are going to use if for general persistent shared files).